crates/shared/src/order_validation.rs

```diff
@@ -339,7 +341,29 @@ impl OrderValidator {
             code_fetcher,
             app_data_validator,
             max_gas_per_order,
-        }
+        };
+
+        banned_users.spawn_maintenance_task(move |cache| async move {
```

Comment:
I think the fact that this is needed in two places is a smell. Why does the callsite need to think about the cache updating logic when it is almost identical to the other callsite?